### PR TITLE
[12.0][MIG] hr_timesheet_sheet

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

File: hr_timesheet_sheet/data/hr_timesheet_sheet_data.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet/data/hr_timesheet_sheet_data.xml]: @alexey-pelykh `<odoo noupdate="1">`

File: hr_timesheet_sheet/views/hr_department_views.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    -->
+
+    <!--Hr Department Inherit Kanban view-->
+    <record id="hr_department_view_kanban" model="ir.ui.view">
+        <field name="name">hr.department.kanban.inherit</field>
+        <field name="model">hr.department</field>
+        <field name="inherit_id" ref="hr.hr_department_view_kanban"/>
+        <field name="groups_id" eval="[(4,ref('hr_timesheet.group_hr_timesheet_user'))]"/>
+        <field name="arch" type="xml">
+            <data>
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet/views/hr_department_views.xml]: remove

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

**Comment by JordiBForgeFlow** [comment]
@alexey-pelykh why do you remove the migration scripts? They are clearly enclosed under a version number, so they are not harmful. If you migrate from v10 to v12 the scripts will be used.

File: hr_timesheet_sheet/views/hr_department_views.xml (hunk 2 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet/views/hr_department_views.xml]: Please elaborate?

**Comment by alexey-pelykh** [comment]
@jbeficent only because of "Remove any possible migration script from previous version." in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

File: hr_timesheet_sheet/views/hr_department_views.xml (hunk 2 above)
Comment (nikul-serpentcs) [reply hr_timesheet_sheet/views/hr_department_views.xml]: @alexey-pelykh IMO `<data>` is not need here, Please remove it!

Comment (alexey-pelykh) [reply hr_timesheet_sheet/views/hr_department_views.xml]: Ah, thanks for clarification!

**Comment by alexey-pelykh** [comment]
@jbeficent @pedrobaeza in case those migration scripts should've been kept despite of the direct rule from the migration guidelines, a clarification should be added to the rules?

**Comment by JordiBForgeFlow** [comment]
@alexey-pelykh @pedrobaeza I do not really understand this rule. I personally prefer ton maintain the modules as equal as possible across versons

**Comment by JordiBForgeFlow** [comment]
@alexey-pelykh oh well it does not make sense to hold a script for an older version if the corresponing module code might not be compatible. So I agree with the rule :)

**Comment by pedrobaeza** [comment]
Jordi, they should be removed because they can interact the bad way when migrating a DB, and as OpenUpgrade requires to go version per version migrating, you don't get any advantage keeping them, only possible drawbacks.

**Comment by alexey-pelykh** [comment]
thanks for clarification! btw, this PR depends on https://github.com/OCA/web/pull/1101

**Comment by alexey-pelykh** [comment]
Summoning good people of 11.0-related PRs that are aware of the matter: @jbeficent @lreficent @mreficent @oihane @astirpe - I'd like to request a review, if you don't mind

**Comment by alexey-pelykh** [comment]
One question to whoever may posses this knowledge:
`sheet_id` and `sheet_id_computed` - why they are separate? (@mreficent)

**Comment by MiquelRForgeFlow** [comment]
That is what odoo had in v10 and I just maintained that.

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by alexey-pelykh** [comment]
I do think that @Icallhimtest (as original author of that line) would be able to bring in some clarity, yet that's not a blocker as of now

File: hr_timesheet_sheet/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    'author': (
+        'Eficent, '
+        'Odoo Community Association (OCA)'
+    ),
+    'website': 'https://github.com/OCA/hr-timesheet',
+    'depends': [
+        'hr_timesheet',
+        'web_widget_x2many_2d_matrix',
+    ],
+    'data': [
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet/__manifest__.py:22]: `data` sequence should be [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L29)
If XML `group` not use CSV then LGTM

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

File: hr_timesheet_sheet/__manifest__.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet/__manifest__.py:22]: Adjusted as per template

Comment (nikul-serpentcs) [reply hr_timesheet_sheet/__manifest__.py:22]: > Adjusted as per template

It's Good :+1:

**Comment by alexey-pelykh** [comment]
@pedrobaeza please merge when you'll have a spare minute

**Pull request merged**